.test-defs/cmd/create-shoot/main.go

```diff
@@ -0,0 +1,146 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/test/integration/framework"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	shootName         string
+	projectNamespace  string
+	kubeconfigPath    string
+	cloudprovider     gardenv1beta1.CloudProvider
+	cloudprofileName  string
+	secretBindingName string
+	region            string
+	zone              string
+	k8sVersion        string
+
+	// optional parameters
+	machineType   string
+	autoScalerMin *int
+	autoScalerMax *int
+
+	testLogger *logrus.Logger
+)
+
+func init() {
+	testLogger = logger.NewLogger("debug")
+
+	shootName = os.Getenv("SHOOT_NAME")
+	if shootName == "" {
+		testLogger.Fatalf("EnvVar 'SHOOT_NAME' needs to be specified")
+	}
+	projectNamespace = os.Getenv("PROJECT_NAMESPACE")
+	if projectNamespace == "" {
+		testLogger.Fatalf("EnvVar 'PROJECT_NAMESPACE' needs to be specified")
+	}
+	kubeconfigPath = os.Getenv("TM_KUBECONFIG_PATH")
+	if kubeconfigPath == "" {
+		testLogger.Fatalf("EnvVar 'TM_KUBECONFIG_PATH' needs to be specified")
+	}
+	cloudprovider = gardenv1beta1.CloudProvider(os.Getenv("CLOUDPROVIDER"))
+	if cloudprovider == "" {
+		testLogger.Fatalf("EnvVar 'CLOUDPROVIDER' needs to be specified")
+	}
+	cloudprofileName = os.Getenv("CLOUDPROFILE")
+	if cloudprofileName == "" {
+		testLogger.Fatalf("EnvVar 'CLOUDPROFILE' needs to be specified")
+	}
+	secretBindingName = os.Getenv("SECRET_BINDING")
+	if secretBindingName == "" {
+		testLogger.Fatalf("EnvVar 'SECRET_BINDING' needs to be specified")
+	}
+	region = os.Getenv("REGION")
+	if region == "" {
+		testLogger.Fatalf("EnvVar 'REGION' needs to be specified")
+	}
+	zone = os.Getenv("ZONE")
+	if zone == "" && cloudprovider != gardenv1beta1.CloudProviderAzure {
+		testLogger.Fatalf("EnvVar 'ZONE' needs to be specified")
+	}
+	k8sVersion = os.Getenv("K8S_VERSION")
+	if k8sVersion == "" {
+		testLogger.Fatalf("EnvVar 'K8S_VERSION' needs to be specified")
+	}
+
+	machineType = os.Getenv("MACHINE_TYPE")
+	if autoScalerMaxEnv := os.Getenv("AUTOSCALER_MIN"); autoScalerMaxEnv != "" {
+		autoScalerMaxInt, err := strconv.Atoi(autoScalerMaxEnv)
+		if err != nil {
+			testLogger.Infof("Cannot parse %s: %s", autoScalerMaxEnv, err.Error())
+		}
+		autoScalerMin = &autoScalerMaxInt
+	}
+	if autoScalerMinEnv := os.Getenv("AUTOSCALER_MAX"); autoScalerMinEnv != "" {
```

Comment:
I'm still confused. The variable name is `autoscalerMax`, but you read `AUTOSCALER_MIN` from env?!